src/controllers/payment.controller.js

```diff
@@ -0,0 +1,99 @@
+import {
+  stripeToken,
+  paymentMethod,
+  charge,
+  stripeListener,
+  sendEmails,
+} from '../utils';
+import { orderService } from '../services';
+import { stripe } from '../helpers';
+
+const { ADMIN_EMAIL } = process.env;
+const makePayment = async (req, res) => {
+  const { orderId } = req.params;
+  const token = await stripeToken(req.body);
+  const method = await paymentMethod(token.id);
+  const amount = req.orderAmount;
+  const chargeBody = {
+    amount,
+    method: method.id,
+    orderId,
+    userId: req.user.id,
+  };
+  const payment = await charge(chargeBody);
+  if (
+    payment.status === 'requires_action' &&
+    payment.next_action.type === 'redirect_to_url'
+  ) {
+    const redirectUrl = payment.next_action.redirect_to_url.url;
+    return res.status(401).json({
+      code: '401',
+      Message: 'Additional action required',
+      Action: redirectUrl,
+    });
+  }
+  return res
+    .status(200)
+    .json({ code: '200', Message: 'Payment succedded', payment });
+};
+
+const webhook = async (req) => {
+  try {
+    const body = JSON.stringify(req.body, null, 2);
+    const secret = process.env.WEBHOOK_SECRET;
+    const header = stripe.webhooks.generateTestHeaderString({
+      payload: body,
+      secret,
+    });
+    const event = stripe.webhooks.constructEvent(body, header, secret);
+    const data = await stripeListener(event);
+    if (data) {
+      if (data.status === 'succeeded') {
+        const status = 'succeeded';
+        const update = await orderService.updateOrderStatus(
+          status,
+          data.metadata.orderId
+        );
+        return update;
+      }
+      if (data.status === 'requires_action') {
+        const status = 'incomplete';
+        const update = await orderService.updateOrderStatus(
+          status,
+          data.metadata.orderId
+        );
+        return update;
+      }
+      if (data.status === 'requires_payment_method') {
+        const status = 'failed';
+        const update = await orderService.updateOrderStatus(
```

Comment:
The updateOrderStatus function is called multiple times in the webhook function with different status values. It would be better to create a separate function for updating the order status and reuse it. As in separate functions to update the to different statuses i.e: succeeded, incomplete and failed so that they can be independently reusable too.